src/condor_daemon_core.V6/daemon_core.cpp

```diff
@@ -11383,3 +11419,105 @@ int DaemonCore::CreateProcessNew(
 	if(! ms.empty()) { ocpa.err_return_msg = ms; }
 	return rv;
 }
+
+
+bool
+DaemonCore::SetupAdministratorSession(unsigned duration, std::string &capability)
+{
+	if (!m_enable_remote_admin) {
+		return false;
+	}
+
+		// Add some modest caching to the admin session; prevents us from
+		// using a unique session per d-slot in the startd.
+	time_t now = time(NULL);
+	if (m_remote_admin_last_time + 30 > now) {
+		capability = m_remote_admin_last;
+		return true;
+	}
+	if (duration < 30) {duration = 30;}
+
+	m_remote_admin_seq++;
+
+	std::string id;
+	formatstr( id, "%s#%ld#%lu", daemonCore->publicNetworkIpAddr(),
```

Comment:
These session ids collide with the ones the startd constructs for match claiming. They need to be altered to be unique.